File: data-structures/queue/problems/ts/leetcode-950.ts
function deckRevealedIncreasing(deck: number[]): number[] {
    deck.sort((a, b) => a - b)
    const n = deck.length
    const result = new Array(n)
    const queue = Array.from({ length: n }, (_, i) => i);

    for(const card of deck) {
        let idx = queue.shift() as number
        result[idx] = card
        if(queue.length > 0) {
            queue.push(queue.shift() as number)
        }
    }

   


    return result
    
};


console.log(deckRevealedIncreasing([17,13,11,2,3,5,7]))

File: data-structures/queue/problems/java/leetcode-950.java
import java.util.Queue;
import java.util.ArrayDeque;
import java.util.Arrays;

class Solution {

    public int[] deckRevealedIncreasing(int[] deck) {
        int n = deck.length;
        Queue<Integer> queue = new ArrayDeque<Integer>();
        for (int i = 0; i < n; i++) {
            queue.add(i);
        }
        int[] res = new int[n];
        Arrays.sort(deck);

        for (int card : deck) {
            int idx = queue.remove();
            res[idx] = card;

            if (queue.size() > 0) {
                queue.add(queue.poll());
            }
        }
        return res;

    }
}